src/func/imports/create-import.js

```diff
@@ -107,25 +109,43 @@ export function createImport(orm, importData) {
 			]);
 
 		// Create entityTypedataId
-		const [dataId] = await createImportDataRecord(
-			transacting,
-			camelToSnake({
-				aliasSetId: aliasSet && aliasSet.get('id'),
-				disambiguationId:
-				disambiguationObj && disambiguationObj.get('id'),
-				identifierSetId: identifierSet && identifierSet.get('id'),
-				...entityDataSets
-			}),
-			importData
-		);
+		let dataId = null;
+		try {
+			[dataId] = await createImportDataRecord(
+				transacting,
+				camelToSnake({
+					aliasSetId: aliasSet && aliasSet.get('id'),
+					disambiguationId:
+					disambiguationObj && disambiguationObj.get('id'),
+					identifierSetId: identifierSet && identifierSet.get('id'),
+					...entityDataSets
+				}),
+				importData
+			);
+		}
+		catch (err) {
+			throw new Error(`Error during dataId creation ${err}`);
+		}
 
 		// Create import entity
-		const [importId] =
-		await createImportRecord(transacting, [{type: entityType}]);
+		let importId = null;
+		try {
+			[importId] =
+				await createImportRecord(transacting, [{type: entityType}]);
+		}
+		catch (err) {
+			throw new Error(`Error during creation of importId ${err}`);
+		}
 
 		// Get origin_source
-		const originSourceId =
-			await getOriginSourceId(transacting, source);
+		let originSourceId = null;
```

Comment:
None of these variables appear to ever be re-assigned. Can you make them const, assigned in the try...catch blocks?

Reply:
They are not being reassigned, but being used again later out side the try...catch blocks. So I can't see anyway of instead using const for them.